contracts/dnsregistrar/OffchainDNSResolver.sol

```diff
@@ -168,17 +173,10 @@ contract OffchainDNSResolver is IExtendedResolver, IERC165 {
             );
     }
 
-    function stripWildcard(
+    function checkWildcard(
         bytes memory name
-    ) public pure returns (bytes memory) {
-        if (name.length > 4 && name[0] == "*" && name[1] == ".") {
-            bytes memory strippedName = new bytes(name.length - 2);
-            for (uint i = 2; i < name.length; i++) {
-                strippedName[i - 2] = name[i];
-            }
-            return strippedName;
-        }
-        return name;
+    ) public pure returns (bool isWildcard) {
+        return name.length > 4 && uint8(name[0]) == 1 && name[1] == "*";
```

Comment:
```suggestion
        return name.length => 4 && uint8(name[0]) == 1 && name[1] == "*";
```

Reply:
isn't minimum valid dns encoded name length 5? i.e. `\*.x.`

Reply:
You're right, I forgot about the trailing \0!